### PR TITLE
Fix the `multiple_runners` test.

### PR DESCRIPTION
**Pull request by teymour-aldridge**
## 🌟 What is the purpose of this PR?

Fixes the `multiple_runners` test.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1199548034582004/1202815168504002) _(internal)_

## 🔍 What does this change?

- Fixes a bug which would happen when trying to use Python and JavaScript runners together.

## 🛡 What tests cover this?

- The test suite (and specifically the `multiple_runners` test)

### CODE REVIEW

File: packages/engine/lib/execution/src/package/simulation/state/behavior_execution/reset_index_col.rs [hunk 1]
```diff
@@ -10,6 +10,8 @@ pub fn reset_index_col(behavior_index_col_index: usize) -> Result<StateColumn> {
     })))
 }
 
+/// This struct is responsible to resetting the value of the behaviour id index column to 0 (_not_
+/// the behaviour list).
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/execution/src/package/simulation/state/behavior_execution/reset_index_col.rs:14]: Can you clarify this a little more? I don't quite understand it

File: packages/engine/lib/execution/src/runner/javascript/mod.rs [hunk 2]
```diff
@@ -1358,26 +1358,20 @@ impl<'s> ThreadLocalRunner<'s> {
         Ok(())
     }
 
+    /// "Flushes" the changes which the JavaScript code made. This involves collecting a list of all
+    /// the changes, which we then use to modify the underlying arrow arrays.
+    ///
+    /// See also the [`memory::arrow::flush`] module for more information.
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/execution/src/runner/javascript/mod.rs:1364]: Appreciate the drive-by doc improvements, thank you!

**Review by Alfred-Mountfield — COMMENTED** [review]
Generally looks good to me :) I think @TimDiekmann should have a look at this one too though

File: packages/engine/lib/execution/src/runner/javascript/mod.rs [hunk 3]
```diff
@@ -1651,6 +1649,18 @@ impl<'s> ThreadLocalRunner<'s> {
...
     }
 
+    /// Reloads the data in the shared_store if necessary.
+    fn reload_data_if_necessary(shared_store: &mut TaskSharedStore) -> Result<()> {
+        let (write_proxies, _) = match shared_store.get_write_proxies() {
+            Ok(t) => t,
+            Err(_) => return Ok(()),
+        };
+        write_proxies.maybe_reload().map_err(|_| {
+            JavaScriptError::Unique("could not reload batches (this is a bug)".to_string())
```
Comment (TimDiekmann) [inline packages/engine/lib/execution/src/runner/javascript/mod.rs:1659]:
```suggestion
            JavaScriptError::from("could not reload batches (this is a bug)")
```

File: packages/engine/lib/execution/src/task/shared_store.rs [hunk 4]
```diff
@@ -29,6 +29,46 @@ impl TaskSharedStore {
...
+}
+
+/// An error encountered when calling [`TaskSharedStore::get_write_proxies`].
+pub enum GetWriteProxiesError {
+    StateNotWritable,
+}
+
+impl From<GetWriteProxiesError> for crate::runner::JavaScriptError {
+    fn from(_: GetWriteProxiesError) -> crate::runner::JavaScriptError {
+        crate::runner::JavaScriptError::Unique("cannot obtain the state as writable".to_owned())
```
Comment (TimDiekmann) [inline packages/engine/lib/execution/src/task/shared_store.rs:70]:
```suggestion
        crate::runner::JavaScriptError::from("cannot obtain the state as writable")
```

**Review by TimDiekmann — CHANGES_REQUESTED** [review]
LGTM, thank you!

Only have two minor things on creating the error.

**Review by TimDiekmann — APPROVED** [review]
🎉

**Pull request merged**